### PR TITLE
Homebrewhighs

### PR DESCRIPTION
**Pull request by donbright**
this is a fix for issue #2219 

basically homebrew is broken on two things. first, libxml requires pkg-config to look in homebrew,

second, homebrew's qscintilla2 is not found by current openscad pri / prf files.

this is repaired by altering the process so that homebrew will now need its own 'setenv' script, like the other builds,  and users will have to manually run it, instead of homebrew being automatically detected like before. it incorporates @kintel suggestion of setting PKG_CONFIG_PATH but now that is just part of the "standard process" of a homebrew build.

IMHO this is a little bit 'cleaner', for example there is no -I/usr/include anywhere in the Makefile so its hopefully not trying to pull in system versions of packages like libxml2.  and it  allows quicker patching of future homebrew issues, for example if new dependencies require new "PKG_CONFIG_PATH" alterations we can just update the setenv_homebrew.sh script

(that mpfr fix sneaked in there too, my bad, shouldnt hurt anything tho)

### CODE REVIEW

**Comment by kintel** [comment]
@donbright Looks like this broke finding boost on homebrew. I haven't looked into the details.

**Comment by t-paul** [comment]
I think it's the CGAL 4.12 update on Travis which broke that.

I hacked around that problem in master with https://github.com/openscad/openscad/commit/a24ecaec107206e1523a97910db4dab152bb578b - I don't have a working env to find a proper fix.

**Comment by kintel** [comment]
@t-paul thx - I've merged with master -let's see...

**Comment by kintel** [comment]
@donbright Looking at this more closely, it looks like the problem is that the new `source scripts/setenv_homebrew.sh` isn't run in the Travis environment.

**Comment by t-paul** [comment]
Yeah, the issue I mentioned above did hit the test suite / cmake build, the qmake application build was ok.

**Comment by donbright** [comment]
dang sorry, i forgot that travis did OSX now

**Comment by donbright** [comment]
i added a fix or two for test build... 

this btw might also fix #2251  although i am on a slightly  newer high sierra IIRC

**Comment by donbright** [comment]
ok so i patched a few things and now it builds on travis... and the tests pass (!)

interesting that on my actual machine a bunch of the tests fail ....

File: scintilla.pri [hunk 1]
```diff
@@ -18,6 +18,13 @@ scintilla {
     }
   }
 
+  BREW=$$(OPENSCAD_HOMEBREW)
+  !isEmpty(BREW) {
+    include(/usr/local/opt/qscintilla2//data/mkspecs/features/qscintilla2.prf)
```
Comment (kintel) [inline scintilla.pri]: Below there's a comment that "qscintilla2.prf which ships with QScintilla is broken for Mac", but here we include it nevertheless.
The broken aspect appears to be that they try to link with `-lqscintilla2_qt5_debug` in debug mode, but don't ship that library.

**Pull request closed without merging**